src/app/authentication/forgot-password/forgot-password.component.ts

```diff
@@ -44,17 +44,15 @@ export class ForgotPasswordComponent {
                             return this.popupService.new({
                                 type: 'error',
                                 title: 'Error changing password',
-                                text: `User did not use email authentication method`
+                                text: 'User did not use email authentication method'
                             }).pipe(switchMap(() => EMPTY));
                         }
                 }
                 return throwError(err);
             }),
             displayBackendErrorRx())
-            .pipe(switchMap(() => this.popupService.new({
-                type: 'success',
-                title: 'Success',
-                text: 'We have sent you an e-mail containing your password reset link.'
-            }).pipe(switchMap(() => this.router.navigate(['/'])))));
+            .pipe(switchMap(() => this.popupService.success(
+                'We have sent you an e-mail containing your password reset link.'
+            ).pipe(switchMap(() => this.router.navigate(['/'])))));
```

Comment:
Sorry, I didn't see this code from Marko before. There is no need for additional pipes. Just put those `switchMap`s after `displayBackendErrorRx()`. Something like:

```
return this.signUpService.forgotPassword(email).pipe(
  catchError(() => // custom error handling),
  displayBackendErrorRx(),
  switchMap(() => // popup),
  tap(() => // navigate)
)
```

Additional note, it's good practice to put navigation to `tap` operator because we are not expecting the result from navigation promise; if it fails - it fails, doesn't matter. the router will take care of it.